File: PROVA_TRIM2_EDUARDA/programa2.py

def determinar_estacao(dia, mes):
    if (mes == 9 and dia >= 23) or mes == 10 or mes == 11 or (mes == 12 and dia < 22):
        return "Primavera"
    elif (mes == 12 and dia >= 22) or mes == 1 or mes == 2 or (mes == 3 and dia < 21):
        return "Verão"
    elif (mes == 3 and dia >= 22) or mes == 4 or mes == 5 or (mes == 6 and dia < 21):
        return "Outono"
    else:
        return "Inverno"

def main():
    dia = int(input("Digite o dia: "))
    mes = int(input("Digite o mês (em número): "))

    if mes < 1 or mes > 12 or dia < 1 or dia > 31:
        print("Data inválida.")
    else:
        estacao = determinar_estacao(dia, mes)
        print(f"Na data {dia}/{mes}, estamos na estação: {estacao}")

if __name__ == "__main__":
    main()

File: PROVA_TRIM2_EDUARDA/programa2.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function determinarEstacao(dia, mes) {

  if ((mes === 9 && dia >= 23) || (mes === 10) || (mes === 11) || (mes === 12 && dia < 22)) {
    return "Primavera";
  } else if ((mes === 12 && dia >= 22) || (mes === 1) || (mes === 2) || (mes === 3 && dia < 21)) {
    return "Verão";
  } else if ((mes === 3 && dia >= 22) || (mes === 4) || (mes === 5) || (mes === 6 && dia < 21)) {
    return "Outono";
  } else {
    return "Inverno";
  }
}

rl.question('Digite o dia: ', (dia) => {
  rl.question('Digite o mês (em número): ', (mes) => {
    dia = parseInt(dia);
    mes = parseInt(mes);

    if (isNaN(dia) || isNaN(mes) || mes < 1 || mes > 12 || dia < 1 || dia > 31) {
      console.log('Data inválida.');
    } else {
      const estacao = determinarEstacao(dia, mes);
      console.log(`Na data ${dia}/${mes}, estamos na estação: ${estacao}`);
    }

    rl.close();
  });
});
